src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,284 @@
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        ppas.append(ppa)
+        pals.append(pal)
+        # ESC senosrs
+        esc_sensors = []
+        esc_sensors.append(json.load(
+              open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json'))))
+        config = {
+            'registrationRequests': devices,
+            'conditionalRegistrationData': conditionals,
+            'grants': grants,
+            'ppas': ppas,
+            'pals': pals,
+            'esc_sensors' : esc_sensors
+        }
+        writeConfig(filename, config)
+    def assertEqualToDeviceParamOrIfExistWithDefault(self, attr_name, parent_attr_in_record,\
+                                          parent_attr_in_device, attr_defaule_value):
+        if attr_name in parent_attr_in_device:
+           self.assertEqual(parent_attr_in_device[attr_name],\
+             parent_attr_in_record[attr_name])
+        elif attr_name in parent_attr_in_record:
+            self.assertEqual(parent_attr_in_record[attr_name], attr_defaule_value)
+        
+    def assertCbsdRecord(self, registration_request, grant_request, grant_response, cbsd_dump_data):
+        for index, device in enumerate(registration_request):
+            record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+            cbsd_record = [record['registration'] for record in cbsd_dump_data if record['id'] == record_id]
+            self.assertEqual(1, len(cbsd_record))
+            # required parameters
+            self.assertEqual(device['cbsdCategory'], cbsd_record[0]['cbsdCategory'])
+            self.assertEqual(device['fccId'], cbsd_record[0]['fccId'])
+            self.assertEqual(device['airInterface']['radioTechnology'],\
+                             cbsd_record[0]['airInterface']['radioTechnology'])
+            self.assertEqual(device['installationParam']['latitude'],\
+                             cbsd_record[0]['installationParam']['latitude'])
+            self.assertEqual(device['installationParam']['longitude'],\
+                             cbsd_record[0]['installationParam']['longitude'])
+            self.assertEqual(device['installationParam']['height'],\
+                             cbsd_record[0]['installationParam']['height'])
+            self.assertEqual(device['installationParam']['heightType'], \
+                             cbsd_record[0]['installationParam']['heightType'])
+            self.assertEqual(device['installationParam']['antennaGain'], \
+                             cbsd_record[0]['installationParam']['antennaGain'])
+             # parameters should exist in record if exist in device,\
+             # if exist only in record, its value should be equal to a default value          
```

Comment:
I think the default values exist implicitly otherwise how the SAS will compute propagation of the CBSD which doesn't provide beamwidth, downtilt or antennaAzimuth.
and if they are not in the dump how will other SASs calculate the propagattion of those CBSDs? it will use default values, won't it?

Reply:
The SAS will use default values when doing the actual computations, but it's my understanding that the SAS should not fill the default values when exchanging info with another SAS (although that may be a good idea, I don't think it's allowed based on the part of the spec I quoted above).